File: 240229/선두를 지켜라 3/keep-the-lead-3.py
N, M = map(int, input().split())

def gen_pos_per_sec(cmd_arr):
    pos = [0]

    for v, t in cmd_arr:
        pos += list(range(pos[-1] + v, pos[-1] + v * t + 1, v))

    return pos

A_pos = gen_pos_per_sec([tuple(map(int, input().split())) for _ in range(N)])
B_pos = gen_pos_per_sec([tuple(map(int, input().split())) for _ in range(M)])

ans = 0

for i in range(1, len(A_pos)):
    gap = lambda x : A_pos[x] - B_pos[x]

    if gap(i) != 0: # 양수거나, 음수거나. 바뀌었단건, 이전과 곱했을 때 0이거나 음수
        if gap(i) * gap(i - 1) <= 0:
            ans += 1
    
    else: # 0이란 것. 이전에 0이 아니기만 하면 됨
        if gap(i - 1) != 0:
            ans += 1

print(ans)

File: 240229/선두를 지켜라 3/keep-the-lead-3.c
#include <stdio.h>
#define SIZE (1000*1000)
int a_pos[SIZE], b_pos[SIZE];

int main() {
    int n, m, v, t, cur = 0, ans = 0, i;

    scanf("%d %d", &n, &m);

    // 배열 초기화
    a_pos[0] = 0;
    b_pos[0] = 0;

    while(n--) {
        scanf("%d %d", &v, &t);

        for(i = cur + 1; i <= cur + t && i < SIZE; i++) // SIZE를 초과하지 않도록 보호
            a_pos[i] = a_pos[i - 1] + v;

        cur = i - 1;
    }

    cur = 0;

    while(m--) {
        scanf("%d %d", &v, &t);

        for(i = cur + 1; i <= cur + t && i < SIZE; i++) { // SIZE를 초과하지 않도록 보호
            b_pos[i] = b_pos[i - 1] + v;
        }

        cur = i - 1;
    }

for (int i = 1; i < cur; ++i) {
        int gap = a_pos[i] - b_pos[i];
        
        if (gap != 0) {
            if (i == 1 || gap * (a_pos[i-1] - b_pos[i-1]) <= 0) {
                ans++;
            }
        } else if (a_pos[i-1] - b_pos[i-1] != 0) {
            ans++;
        }
    }

    printf("%d", ans);

    return 0;
}